include/convex_bodies/ballintersectconvex.h

```diff
@@ -130,6 +135,77 @@ class BallIntersectPolytope {
         return std::pair<NT, int>(std::min(polypair.first, ball_lambda.first), facet);
     }
 
+    //---------------------new billiard---------------------//
+    template <typename update_parameters>
+    std::pair<NT, int> line_first_positive_intersect(PointType const& r,
+                                                     PointType const& v,
+                                                     VT& Ar,
+                                                     VT& Av,
+                                                     update_parameters &params) const
+    {
+        std::pair <NT, int> polypair = P.line_first_positive_intersect(r, v, Ar, Av, params);
+        std::pair <NT, int> ball_lambda = B.line_positive_intersect(r, v);
+        int facet = P.num_of_hyperplanes();
+        params.facet_prev = polypair.second;
+
+        if (polypair.first < ball_lambda.first ) {
+            facet = polypair.second;
+            params.hit_ball = false;
+        } else {
+            params.hit_ball = true;
+        }
+
+        return std::pair<NT, int>(std::min(polypair.first, ball_lambda.first), facet);
+    }
+
+    template <typename update_parameters>
+    std::pair<NT, int> line_positive_intersect(PointType const& r,
+                                               PointType const& v,
+                                               VT& Ar,
+                                               VT& Av,
+                                               NT const& lambda_prev,
+                                               MT const& AA,
+                                               update_parameters &params) const
+    {
+        std::pair <NT, int> polypair = P.line_positive_intersect(r, v, Ar, Av, lambda_prev, AA, params);
+        std::pair <NT, int> ball_lambda = B.line_positive_intersect(r, v);
+        int facet = P.num_of_hyperplanes();
+        params.facet_prev = polypair.second;
+
+        if (polypair.first < ball_lambda.first ) {
+            facet = polypair.second;
+            params.hit_ball = false;
+        } else {
+            params.hit_ball = true;
+        }
+
+        return std::pair<NT, int>(std::min(polypair.first, ball_lambda.first), facet);
+    }
+
+    template <typename update_parameters>
+    std::pair<NT, int> line_positive_intersect(PointType const& r,
+                                               PointType const& v,
+                                               VT& Ar,
+                                               VT& Av,
+                                               NT const& lambda_prev,
+                                               update_parameters &params) const
+    {
+        std::pair <NT, int> polypair = P.line_positive_intersect(r, v, Ar, Av, lambda_prev, params);
```

Comment:
Please check now

Reply:
Better now. Instead of 
```
int facet = P.num_of_hyperplanes();

        if (polypair.first < ball_lambda.first ) facet = polypair.second;
```
why not just 
`int facet = polypair.first < ball_lambda.first ?  polypair.second : P.num_of_hyperplanes();`

Reply:
thanks, done